src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int factCapacity;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        factCapacity = INITIAL_CAPACITY;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private int hash;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.hash = hash;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        checkSize();
+        int hash = hash(key);
+        int bucketIndex = hash % factCapacity;
+
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K, V> tmp = table[bucketIndex];
```

Comment:
```suggestion
            Node<K, V> node = table[bucketIndex];
```

This name is better than the abbreviation tmp